### DISCUSSION TITLE
Can not install Truffle Suite directly from extension (Truffle for VS Code) in VS Code and Ganache also

### DISCUSSION THREAD

**Discussion by blockchaingeneration** [ConsenSys-archive/truffle#5606] [General Q&A]
Dear community,
 
I can not install lest version of Truffle Suite in VS Code and Ganache also.
I got this after I typed: npm install -g truffle@latest

 
npm ERR! code EACCES
npm ERR! syscall mkdir
npm ERR! path /usr/local/lib/node_modules/truffle
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/truffle'
npm ERR!  [Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/truffle'] {
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'mkdir',
npm ERR!   path: '/usr/local/lib/node_modules/truffle'
npm ERR! }
npm ERR! 
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR! 
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!

Anybody could help me what am I doing wrong?
I am using macOS Moneterey 12.6.

**Comment by cipaterson** [comment]
Hi,
have you read the advice here:
https://trufflesuite.com/docs/vscode-ext/installation-guide/

You are using macOS, have you installed node with a version manager such as "nvm"?  That is the cleanest way to avoid any permission (EACCES) errors while installing npm packages.

Warm regards,
Chris | Truffle | ConsenSys

**Comment by cds-amal** [comment]
Your `npm` setup has permissions issues (maybe you used `sudo`?). 

`npm doctor` is a useful diagnostic tool that returns the status of your `npm` setup, and hints to fix if necessary.

```
$ npm doctor
npm WARN verifyCachedFiles Content garbage-collected: 632 (188592821 bytes)
npm WARN verifyCachedFiles Cache issues have been fixed
Check                               Value  Recommendation/Notes
npm ping                            ok
npm -v                              ok     current: v8.19.2, latest: v8.19.2
node -v                             ok     current: v18.10.0, recommended: v18.10.0
npm config get registry             ok     using default registry (https://registry.npmjs.org/)
which git                           ok     /usr/bin/git
Perms check on cached files         ok
Perms check on local node_modules   ok
Perms check on global node_modules  ok
Perms check on local bin folder     ok
Perms check on global bin folder    ok
Verify cache contents               ok     verified 9403 tarballs
```

**Comment by blockchaingeneration** [comment]
Thanks for everybody, my problem is solved :)
